### ISSUE TITLE
SWC Plugin?

### ISSUE THREAD

**Issue by rastertail** [ryansolid/dom-expressions#10]
[SWC](https://swc-project.github.io/) is an alternative to Babel that boasts having much better performance, so it would be nice to see an SWC equivalent to `babel-plugin-jsx-dom-expressions` (and also `babel-preset-solid`, though that's technically not relevant here :smile:) in order to speed up the development cycle. However, I haven't looked into the similarities/differences between the two plugin APIs, so I'm not sure how much of a project this would be.

**Comment by ryansolid** [comment]
Cool I haven't seen that yet. The plugins look similar but not the same. Different syntax and path following organization. Seems like it could be possible, mind you it would also take some doing as I built the babel plugin only ever assuming babel as a target so it isn't setup to just pickup and move. Or serve multiple parsers. Definitely consideration when I rewrite and generalize things.

**Comment by MrFoxPro** [comment]
Yeah it would be awesome to implement transformer for ts-loader. I found that loader for babel can be rewritten for ts-loader, as there are many similar things.

**Comment by feledori** [comment]
As swc now released their wasm plugin api this would be interesting. As far as I read discussions on solidjs this would enable solid on deno. Not really capable of writing a plugin just wanted to bring up the discussion again.😄 cheers

**Comment by modderme123** [comment]
Linking https://github.com/modderme123/swc-plugin-jsx-dom-expressions as an initial implementation of the client/dom transform (still super WIP, contributions are welcome)

**Comment by ryansolid** [comment]
The 2.x compiler is Oxc (`@dom-expressions/compiler`), now moving into the Solid repo. Not building an SWC plugin for 1.x.
